src/main/java/io/jenkins/plugins/sprp/YAML_FlowDefinition.java

```diff
@@ -125,6 +138,36 @@ public FlowExecution create(FlowExecutionOwner owner, TaskListener listener,
         return new CpsFlowExecution(script, false, owner);
     }
 
+    private List<String> getCleanRefspecs(List<UserRemoteConfig> userRemoteConfigs){
+        List<String> refSpecs = new ArrayList<>();
+
+        for(UserRemoteConfig urc: userRemoteConfigs)
+            for(String s: urc.getRefspec().split("\\+", 0))
+                if(!s.equals(""))
+                    refSpecs.add(s);
+
+        for(int i = 0; i < refSpecs.size(); i++)
+            refSpecs.set(i, refSpecs.get(i).trim());
+
+        return refSpecs;
+    }
+
+    private String getBranchName(String[] refSpecsArray){
+        boolean done = false;
+        StringBuilder branchName = new StringBuilder();
+        for(int i = 0; i < refSpecsArray.length && !done; i++){
+            if(refSpecsArray[i].equals("upstream") || refSpecsArray[i].equals("origin")){
```

Comment:
I saw bitbucket generated following two refspecs:
refs/heads/abhishekg1128/readmemd-edited-online-with-bitbucket-1529079381853:refs/remotes/origin/PR-1
refs/heads/master:refs/remotes/upstream/master

And GitHub generates
+refs/pull/3/head:refs/remotes/origin/PR-3 +refs/heads/master:refs/remotes/origin/master

Reply:
this kind of info would be a great thing to add as a comment to the function definition

Reply:
Sure I will add this on the top of the function.